### ISSUE TITLE
Exception when opening Settings page

### ISSUE THREAD

**Issue by Germs2004** [gitextensions/gitextensions#2886]
When I open Git Extensions and go to the Settings page, I immediately get the following error. I tried uninstalling & reinstalling, and I tried deleting the settings file in AppData, but I still get the error every time. I think the problem began right after I installed Visual Studio 2015 Community on the computer.

GitExtensions.exe [2.48.05]

The path is not of a legal form.

at System.IO.Path.NormalizePath(String path, Boolean fullCheck, Int32 maxPathLength, Boolean expandShortPaths)
at System.IO.Path.GetFullPathInternal(String path)
at System.IO.Path.GetFullPath(String path)
at System.IO.FileSystemWatcher.StartRaisingEvents()
at GitCommands.Settings.FileSettingsCache.LoadImpl()
at GitCommands.SettingsCache.<>c__DisplayClass1.b__0()
at GitCommands.SettingsCache.LockedActionT
at GitCommands.SettingsCache.<>c__DisplayClass1.b__0()
at GitCommands.SettingsCache.LockedActionT
at GitCommands.SettingsCache.EnsureSettingsAreUpToDate()
at GitCommands.SettingsCache.<>c__DisplayClass151.<TryGetValue>b__14() at GitCommands.SettingsCache.LockedAction[T](Func1 action)
at GitCommands.SettingsCache.TryGetValueT
at GitCommands.Settings.SettingsContainer2.TryGetValue[T](String name, T defaultValue, Func2 decode, T& value)
at GitCommands.Settings.SettingsContainer2.GetValue[T](String name, T defaultValue, Func2 decode)
at GitCommands.Settings.ConfigFileSettings.GetEncoding(String settingName)
at GitUI.CommandsDialogs.SettingsDialog.Pages.GitConfigSettingsPage.SettingsToPage()
at GitUI.CommandsDialogs.SettingsDialog.SettingsPageBase.LoadSettings()
at GitUI.CommandsDialogs.FormSettings.LoadSettings()
at GitUI.CommandsDialogs.FormSettings.FormSettings_Shown(Object sender, EventArgs e)
at System.Windows.Forms.Form.OnShown(EventArgs e)
at System.Windows.Forms.Control.InvokeMarshaledCallbackHelper(Object obj)
at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state, Boolean preserveSyncCtx)
at System.Threading.ExecutionContext.Run(ExecutionContext executionContext, ContextCallback callback, Object state, Boolean preserveSyncCtx)
at System.Threading.ExecutionContext.Run(ExecutionContext executionContext, ContextCallback callback, Object state)
at System.Windows.Forms.Control.InvokeMarshaledCallback(ThreadMethodEntry tme)
at System.Windows.Forms.Control.InvokeMarshaledCallbacks()

**Comment by Germs2004** [comment]
If I open Git Extensions from the start menu, I get the error.  If I am in Visual Studio and click from the menu GitExt > Commit, then it immediately opens the Settings menu without errors.  There, I can click Cancel and it goes to the Commit page.

**Comment by atjshop** [comment]
I have exact same issue. Upgraded to the newest version, still same error

**Comment by DCBlaze** [comment]
I had the same error after configurating GitExtensions and reopening the settings-dialog. First I've tried to remove the settings files in AppData & my home dir. Both wasn't successful. Opening the settings dialog via cmd (gitex settings) worked fine as long as I havn't switched to the GitExtensions directory. Now I found something strange:
After removing or renaming the file "config" (without any extension) in the GitExtensions-directory, the settings dialog will open again. The strange thing is, that the only thing this file contains is a language entry for the "filesEncoding". Has anyone an idea which setting this file will generate? (Seems not to be the default file encoding...)

**Comment by X-Coder** [comment]
I had exactly the same issue as DCBlaze. In my case, some setting had to be missing in some config file, so the config-dialog auto opened each time I launched gitextensions and crashed immediately, preventing me from setting it up or use it.

Uninstalling/Repair didn't delete the "config" file. After deleting the "config" file manually in the installation folder, I could setup the missing settings. And the crashes were gone.

In my config file there was only this text:
[user]
	name = My Username
	email = myemail@example.com

**Comment by geertdoornbos** [comment]
Experienced same issue. Also deleted the config file in the installation folder:  C:\Program Files (x86)\GitExtensions

The config file should not be created / located in the installation folder. May be it's placed there while executing GitExtensions 'as administrator' (?)

**Comment by vbjay** [comment]
The config file is needed.  That is how .net stores some config things for
your exe.  The settings for the program are stored in another file.  Based
on the isportable setting in the config file, the settings are stored with
the exe or in the user's profile.

**Comment by geertdoornbos** [comment]
@vbjay The .NET config file is GitExtensions.exe.config. I meant (and also @X-Coder I think) a 'config' file. Without extension.

My 'config' file only contained a reference to a .ppk file.

**Comment by vbjay** [comment]
Right.  That file should not be deleted.  The settings file
is GitExtensions.settings which is located at
%appdata%\\GitExtensions\GitExtensions  or with the program.  The
isportable setting controls wich file is read.

**Comment by X-Coder** [comment]
I was talking about the "config" file without extension in Program Files (x86)\GitExtensions folder.

##### Steps to reproduce:
Create a new "config" file in Program Files (x86)\GitExtensions folder, with just this content:
```
[user]
name = My Username
email = myemail@example.com
```
Then Open GitExtensions and go into settings.

It will instantly crash with the error Germs2004 posted above. If some required setting is missing which triggers the settings-dialog to open up on start GitExtensions crashes right on start, second one could happen on updates or (re-)installation.

**Comment by geertdoornbos** [comment]
@vbjay Sorry, don't know if I understand you correctly. The issue is about the file:

C:\Program Files (x86)\GitExtensions\config

Not the settings file, not the .NET .config file.

**Comment by vbjay** [comment]
That file or folder does not exist on my machine.  Sounds like you can get
rid of that safely.  I wrote the portability code so I am familiar with
where the files are stored.

**Comment by X-Coder** [comment]
Whats the purpose of this file?
If it is not needed why it is being parsed by GitExtensions?

**Comment by vbjay** [comment]
Show the contents.

**Comment by X-Coder** [comment]
The content seems to differ. In my case it was just:
```
[user]
name = My Username
email = myemail@example.com
```
@geertdoornbos said his file contained a reference to a .ppk file. And @DCBlaze's one contained just a language entry for the "filesEncoding".

**Comment by vbjay** [comment]
Ok.  That looks like a git config file.  Which is certainly safe to remove
in that path.  git config files exist in ~/.gitconfig  and repo folder
 ~ being user's profile path.  .

**Comment by jbialobr** [comment]
Fixed in 01c252aa7edf27b508a77c074a7839c3d713b155

**Comment by puransp** [comment]
Same issue with me. Was not able to open getextension settings or file history etc.
Solved by adding full control permission for current user to folder _"C:\Program Files (x86)\GitExtensions"_ and everything started working for me.
